### PR TITLE
feat: extending v2 with did and dns-did

### PR DESCRIPTION
**Pull request by yehjxraymond**

### CODE REVIEW

File: src/schema/2.0/schema.json [hunk 1]
```diff
@@ -8,23 +8,39 @@
...
           "description": "Url of the website referencing to document store"
+        },
+        "key": {
+          "type": "string",
+          "description": "Public key associated"
         }
       },
-      "required": ["type", "location"],
-      "additionalProperties": false
+      "required": ["type"]
```
Comment (Nebulis) [inline src/schema/2.0/schema.json:22]: there is a regression here, location is no longer mandatory when type is DNS-TXT and you wont be aware before using verify. 

We can be stricter and use oneOf schema way ? (that should set the type correctly) or do the verification manually? (type will be incorrect)

File: src/schema/2.0/schema.json [hunk 2]
```diff
@@ -8,23 +8,39 @@
...
       "properties": {
+        "id": {
+          "type": "string",
+          "description": "Issuer's id, DID can be used"
+        },
         "name": {
           "type": "string",
           "description": "Issuer's name"
         },
+        "revocation": {
```
Comment (Nebulis) [inline src/schema/2.0/schema.json:35]: I think we said revocation is mandatory ? (but that will introduce a breaking change and will be v3).

Same as above => you wont be aware before using verify.

Comment (yehjxraymond) [reply src/schema/2.0/schema.json:71]: This is a little hard to verify with the schema itself, the problem is we are using the child to verify the parent. One approach will be to define different types of `identity` like `identityWithRevocation` & `identityWithoutRevocation` and then inside them specify the internal structure, but i think that will lead to an even more confusing schema to read. I've went with the approach that this is made optional instead and we will leave to verify to settle it.

File: src/schema/2.0/schema.json (hunk 1 above)
Comment (yehjxraymond) [reply src/schema/2.0/schema.json]: done

**Review by Nebulis — APPROVED** [review]

**Pull request merged**